core/src/main/scala/org/ensime/core/javac/JavaCompiler.scala

```diff
@@ -0,0 +1,178 @@
+package org.ensime.core.javac
+
+import akka.actor.ActorRef
+import akka.event.LoggingAdapter
+import com.sun.source.tree.CompilationUnitTree
+import com.sun.source.util.JavacTask
+import com.sun.source.tree.Scope
+import com.sun.source.util.TreePath
+import com.sun.source.util.Trees
+import java.io.{ File, FileInputStream, IOException, InputStream }
+import java.net.URI
+import java.nio.charset.Charset
+import java.util.Locale
+import javax.lang.model.`type`.TypeMirror
+import javax.tools._
+import org.ensime.api._
+import org.ensime.core.DocSigPair
+import org.ensime.util.ReportHandler
+import org.ensime.util.{ FileUtils, ReportHandler }
+import scala.collection.JavaConversions._
+import scala.concurrent.Future
+import scala.reflect.internal.util.{ BatchSourceFile, RangePosition, SourceFile }
+import scala.tools.nsc.Settings
+import scala.tools.nsc.interactive.CompilerControl
+import scala.tools.nsc.io.AbstractFile
+import scala.tools.nsc.reporters.Reporter
+import scala.tools.refactoring.analysis.GlobalIndexes
+
+class JavaCompiler(
+    val config: EnsimeConfig,
+    val charset: Charset,
+    val indexer: ActorRef,
+    val reportHandler: ReportHandler
+) extends JavaDocFinding with JavaCompletion {
+
+  import scala.concurrent.ExecutionContext.Implicits.{ global => exe }
+
+  val compiler = ToolProvider.getSystemJavaCompiler()
+  private val listener = new JavaDiagnosticListener()
+  private val silencer = new SilencedDiagnosticListener()
+  val fileManager = compiler.getStandardFileManager(listener, null, charset)
+  val cp = config.compileClasspath.mkString(File.pathSeparator)
+
+  def askTypecheckFiles(files: Iterable[SourceFileInfo]): Future[Boolean] = Future {
+    reportHandler.clearAllJavaNotes()
+    typecheckAll(files.map(getJavaFileObject))
+    true
+  }
+
+  def askTypeAtPoint(file: SourceFileInfo, offset: Int): Future[Option[TypeInfo]] = Future {
```

Comment:
similarly for all the other methods. Do they really need to be `Future`s? It would be better if they were just normal methods and if we need to call them asynchronously, we'll do it at invocation rather than baked into the API/impl.